analysea/tide.py

```diff
@@ -113,86 +102,107 @@ def demean_amps_phases(
+    # Split the time series into 6-month chunks (default), otherwise specify
+    for start_date, end_date in zip(date_ranges[:-1], date_ranges[1:]):
+        ts_chunk = ts[start_date:end_date]
+        if constituents is None:
+            constituents = calc_constituents(ts=ts, resample_time=resample_time, lat=lat, **kwargs)
+        result_series.append(reconstruct_chunk(ts_chunk, constituents, **kwargs))
 
-    return surge
+    # Concatenate the processed chunks
+    return pd.concat(result_series)
 
 
 def tide_analysis(
-    ts: pd.Series[float],
-    kwargs: Dict[str, Any],
+    ts: pd.Series[float], resample_time: int = 10, lat: float = 0.0, **kwargs: Optional[Dict[Any, Any]]
 ) -> Tuple[pd.DataFrame, pd.DataFrame, npt.NDArray[Any]]:
-    constituents = calc_constituents(ts=ts, kwargs=kwargs)
-    tidal = utide.reconstruct(ts.index, constituents, verbose=kwargs["verbose"])
-    return pd.DataFrame(data=tidal.h, index=ts.index), ts - tidal.h, constituents
+    constituents = calc_constituents(ts=ts, lat=lat, resample_time=resample_time, **kwargs)
+    tidal = utide.reconstruct(ts.index, constituents, **kwargs)
+    tide = pd.Series(data=tidal.h, index=ts.index)
+    surge = pd.Series(data=ts.iloc[0, :].values - tidal.h, index=ts.index)
+    return tide, surge, constituents
 
 
 def yearly_tide_analysis(
-    h: pd.Series[float], split_period: int, kwargs: Dict[str, Any] = OPTS
+    h: pd.Series[float],
+    resample_time: int = 10,
+    split_period: int = 365,
+    lat: int = 0,
+    **kwargs: Optional[Dict[Any, Any]],
 ) -> Tuple[pd.DataFrame, pd.DataFrame, List[npt.NDArray[Any]], List[int]]:
     log = kwargs.get("verbose", False)
 
     min_time = pd.Timestamp(h.index.min())
     max_time = pd.Timestamp(h.index.max())
     date_ranges = pd.date_range(start=min_time, end=max_time, freq=f"{split_period}D")
 
-    tide = pd.DataFrame([])
-    surge = pd.DataFrame([])
+    tide = []
+    surge = []
     coefs = []
     years = []
 
     for start, end in zip(date_ranges[:-1], date_ranges[1:]):
         signal = h[start:end]
 
-        if completeness(signal) > 70:
-            years.append(start.year)
-            tide_tmp, surge_tmp, coef = tide_analysis(signal, kwargs)
-            coefs.append(coef)
-            if not surge_tmp.empty:
-                surge = pd.concat([surge, surge_tmp])
-            if not tide_tmp.empty:
-                tide = pd.concat([tide, tide_tmp])
+        years.append(start.year)
+        tide_tmp, surge_tmp, coef = tide_analysis(signal, lat=lat, resample_time=resample_time, **kwargs)
+        coefs.append(coef)
+        surge.append(surge_tmp)
+        tide.append(tide_tmp)
 
         if log:
             print(f"  => Analyse year {start.year} ({start}-{end})")
             print(f"   +>  {len(tide)} / {len(h)} records done")
 
-    return tide, surge, coefs, years
+    return pd.concat(tide), pd.concat(surge), coefs, years
```

Comment:
I don't have a clear idea of how `yearly_tide_analysis` is being used, but API wise I am not very keen of returning multiple and heterogenous objects from one function. It's quite error prone and requires that every time you use the function you must check the source code in order to get the right order of the arguments. 

If we do need to return all these objects from one function it is usually a better idea to write a small dataclass that will hold these items. For example:
``` python
from dataclasses import dataclass


@dataclass
class TideAnalysisResults:
    tide: pd.DataFrame
    surge: pd.DataFrame
    coefs: list[dict[str, Any]]
    years: list[int]


def yearly_tide_analysis(...):
    ...
    results = TideAnalysisResults(
        tide=tide,
        surge=surge,
        coefs=coefs,
        years=years,
    )
    return results
```

Reply:
leaving it open for now. I haven't touched it

Reply:
implemented